api/enki.py

```diff
@@ -0,0 +1,542 @@
+from nose.tools import set_trace
+from collections import defaultdict
+import datetime
+import base64
+import os
+import json
+import logging
+import re
+from flask.ext.babel import lazy_gettext as _
+
+from sqlalchemy.orm import contains_eager
+
+from lxml import etree
+
+from authenticator import Authenticator
+
+from config import (
+    Configuration,
+    temp_config,
+)
+
+from circulation import (
+    LoanInfo,
+    FulfillmentInfo,
+    HoldInfo,
+    BaseCirculationAPI
+)
+
+from circulation_exceptions import *
+
+from core.util import LanguageCodes
+from core.util.http import (
+    HTTP,
+    RemoteIntegrationException,
+)
+
+from core.coverage import (
+    BibliographicCoverageProvider,
+    CoverageFailure,
+)
+
+from core.model import (
+    get_one_or_create,
+    Collection,
+    Contributor,
+    DataSource,
+    DeliveryMechanism,
+    LicensePool,
+    Edition,
+    Identifier,
+    Library,
+    Representation,
+    Subject,
+    ExternalIntegration,
+)
+
+from core.metadata_layer import (
+    SubjectData,
+    ContributorData,
+    FormatData,
+    IdentifierData,
+    CirculationData,
+    Metadata,
+    ReplacementPolicy,
+)
+
+from core.monitor import (
+    Monitor,
+    IdentifierSweepMonitor,
+    CollectionMonitor,
+)
+
+from core.opds_import import SimplifiedOPDSLookup
+from core.analytics import Analytics
+
+#TODO: Remove unnecessary imports (once the classes are more or less complete)
+
+class EnkiAPI(BaseCirculationAPI):
+
+    DESCRIPTION = _("Integrate an Enki collection.")
+    SETTINGS = [
+        { "key": Collection.EXTERNAL_ACCOUNT_ID_KEY, "label": _("Library ID") },
+        { "key": ExternalIntegration.URL, "label": _("URL") },
+    ] + BaseCirculationAPI.SETTINGS
+
+    PRODUCTION_BASE_URL = "http://enkilibrary.org/API/"
+    availability_endpoint = "ListAPI"
+    item_endpoint = "ItemAPI"
+    #TODO: get this from configurationsettings. Also make sure the value actually gets to the DB.
+    lib = 1
```

Comment:
What's this? If it's the library ID you should be able to get it from ConfigurationSetting now.